<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- param -->
    <arg name="start_node"          default="0"/>
    <arg name="goal_node"           default="0"/>
    <arg name="local_goal_interval" default="1.0"/>
    <arg name="local_goal_dist"     default="5.0"/>
    <arg name="stop_radius_min"     default="0.75"/>
    <arg name="at_stop_node_radius"     default="0.10"/>
    <arg name="count_start_radius"  default="1.5"/>
    <arg name="skip_timecount"      default="7.0"/>
    <arg name="start_delay" default="10.0"/>
    <!-- subscribe topic -->
    <arg name="checkpoint_topic"     default="/node_edge_map/checkpoint"/>
    <arg name="node_edge_map_topic"  default="/node_edge_map"/>
    <arg name="pose_topic"           default="/current_pose"/>
    <arg name="stop_node_flag_topic" default="/is_stop_node_flag"/>
    <!-- publish topic -->
    <arg name="local_goal_topic"         default="/local_goal"/>
    <arg name="current_checkpoint_topic" default="/current_checkpoint"/>
    <arg name="map_file" default=""/>
    <arg name="checkpoint_file" default=""/>

    <!-- node -->
    <node pkg="rwrc22_localgoal_creator" type="rwrc22_localgoal_creator_node" name="localgoal_creator_node" output="screen" launch-prefix="bash -c 'sleep $(arg start_delay); $0 $@' " >
        <!-- param -->
        <param name="start_node"          value="$(arg start_node)"/>
        <param name="goal_node"           value="$(arg goal_node)"/>
        <param name="local_goal_interval" value="$(arg local_goal_interval)"/>
        <param name="local_goal_dist"     value="$(arg local_goal_dist)"/>
        <param name="stop_radius_min"     value="$(arg stop_radius_min)"/>
        <param name="at_stop_node_radius"     value="$(arg at_stop_node_radius)"/>
        <param name="count_start_radius"  value="$(arg count_start_radius)"/>
        <param name="skip_timecount"      value="$(arg skip_timecount)"/>
        <!-- topic -->
        <remap from="/checkpoint"         to="$(arg checkpoint_topic)"/>
        <remap from="/node_edge_map"      to="$(arg node_edge_map_topic)"/>
        <remap from="/current_pose"       to="$(arg pose_topic)"/>
        <remap from="/is_stop_node_flag"  to="$(arg stop_node_flag_topic)"/>
        <remap from="/local_goal"         to="$(arg local_goal_topic)"/>
        <remap from="/current_checkpoint" to="$(arg current_checkpoint_topic)"/>
    </node>
</launch>
